src/System.Management.Automation/engine/hostifaces/Connection.cs

```diff
@@ -631,6 +631,14 @@ public bool RunspaceIsRemote
         }
 
         /// <summary>
+        /// When true this property indicates that this runspace is associated with an interactive session,
+        /// such as a runspace for the PowerShell console.
+        /// A runspace designated as interactive host means that it is the primary runspace, and when it closes
+        /// any other open runspace on the system should be forcibly closed.
+        /// </summary>
+        public bool IsInteractiveHost { get; set; }
```

Comment:
Talked to Paul offline about this property -- how to make sure this property is set only once so that there won't be a misuse of the property. One feedback is to make it internal for now, so that any follow-up work on this matter between RC and GA won't be a breaking change.

Reply:
And the name of the property perhaps can be changed to `IsDefaultRunspace`, without the `Host` in it, so there is less confusion about Runspace vs. Host.